finat/direct_serendipity.py

```diff
@@ -0,0 +1,371 @@
+import numpy
+
+from finat.finiteelementbase import FiniteElementBase
+from finat.physically_mapped import DirectlyDefinedElement
+from FIAT.reference_element import UFCQuadrilateral
+from FIAT.polynomial_set import mis
+
+import gem
+import sympy
+from finat.sympy2gem import sympy2gem
+
+
+class DirectSerendipity(DirectlyDefinedElement, FiniteElementBase):
+    def __init__(self, cell, degree):
+        assert isinstance(cell, UFCQuadrilateral)
+        # assert degree == 1 or degree == 2
+        self._cell = cell
+        self._degree = degree
+        self.space_dim = 4 if degree == 1 else (self.degree+1)*(self.degree+2)//2 + 2
```

Comment:
No need to save this, just shove it in the `space_dimension` method (as commented below).